src/apps/Permissions/Home/AllPermissions.js

```diff
@@ -0,0 +1,189 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import PropTypes from 'prop-types'
+import {
+  DropDown,
+  GU,
+  IconSearch,
+  TextInput,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import EmptyBlock from '../EmptyBlock'
+import EmptyFilteredPermission from '../EmptyFilteredPermissions'
+import PermissionsView from '../PermissionsView'
+
+const ENTITY_TYPES = ['All entities', 'Accounts', 'Apps']
+
+function AllPermissions({ loading, permissions, onManageRole }) {
+  const [selectedEntityType, setSelectedEntityType] = useState(-1)
+  const [searchTerms, setSearchTerms] = useState('')
+  const { layoutName } = useLayout()
+
+  const filteredPermissions = useMemo(() => {
+    return (
+      permissions
+
+        // Filter by search terms
+        .filter(permission => {
+          const preparedSearchTerms = searchTerms.trim().toLowerCase()
+          if (preparedSearchTerms === '') {
+            return true
+          }
+          const appName = permission.app.name.toLowerCase()
+          const roleName = permission.role.name.toLowerCase()
+          const roleId = permission.role.id.toLowerCase()
+          return (
+            appName.includes(preparedSearchTerms) ||
+            roleName.includes(preparedSearchTerms) ||
+            roleId.includes(preparedSearchTerms)
+          )
+        })
+
+        // Filter by account types
+        .filter(permission => {
+          // accounts
+          if (selectedEntityType === 1) {
+            return permission.entities.some(entity => entity.type !== 'app')
+          }
+
+          // apps
+          if (selectedEntityType === 2) {
+            return permission.entities.some(entity => entity.type === 'app')
+          }
+
+          return true
+        })
+    )
+  }, [searchTerms, permissions, selectedEntityType])
+
+  if (loading) {
+    return <EmptyBlock>Loading permissions…</EmptyBlock>
+  }
+
+  if (permissions.length === 0) {
+    return <EmptyBlock>No permissions have been assigned yet.</EmptyBlock>
+  }
+
+  return (
+    <PermissionsView
+      permissions={filteredPermissions}
+      onManageRole={onManageRole}
+      heading={
+        layoutName === 'large' && (
+          <Heading
+            emptyFilter={Boolean(
+              !filteredPermissions.length && permissions.length
```

Comment:
I'd imagine you'd use Boolean for each, not both no? `!Boolean(filteredPermissions.length) && Boolean(permissions.length)`

Reply:
More of a style thing, but I like the single boolean because it ensures we pass a boolean into the prop, rather than having to make sure each condition is wrapped by `Boolean()`.

Reply:
Ahh this was yours @sohkai , my comment was because @bpierre said his preference is to cast using `Boolean` so I'd figured he'd cast both non boolean checks

Reply:
Ah yes I like `Boolean` but only when needed, otherwise I am generally happy to evaluate falsy values as they are :-) And yes here the reason is because `emptyFilter` expects it.